src/Microsoft.AspNet.Mvc.Razor/Resources.resx

```diff
@@ -200,4 +200,7 @@
 
 @{3} "{4}, {5}"</value>
   </data>
+  <data name="LayoutHasCircularReference" xml:space="preserve">
+    <value>Possible circular layout reference detected when rendering '{0}'. The layout page '{1}' has already been rendered.</value>
```

Comment:
Right, we talked about that one ourselves too. Ultimately we disallow *any* form of circular reference, so I think the error message needs to sound stricter...